### DISCUSSION TITLE
Loading Cookies

### DISCUSSION THREAD

**Discussion by Firerouge** [deepjyoti30/ytmdl#255] [Q&A] [answered]
Youtube-dl (and yt-dlp) support loading cookies to access private content. Is there a way to have ytmdl utilize cookies similarly. 

I'd like to download my liked music auto-playlist. This is found for each user at https://music.youtube.com/playlist?list=LM
This playlist can't be made public/unlisted, and the only way to export the songs from it into another playlist is limited to ~200 songs a day: https://github.com/sayok88/likedmusicyoutube

**Comment by nickexyz** [comment]
It would also be nice to be able to download higher quality music when using cookies with YT premium.

> 255kbps audio is extracted (if available) from YouTube Music when premium cookies are given

https://github.com/yt-dlp/yt-dlp#new-features

**Comment by deepjyoti30** [comment, marked as answer]
I understand that cookies can be passed by using a CLI flag to `yt-dlp`. If that is correct, then the `--ytdl-config PATH` flag can be used from `ytmdl` to pass the cookie flag through a file and it will be forwarded to `yt-dlp`.

@Firerouge Let me know if this works. @nickexyz You can confirm as well. If this doesn't work out of the box, you guys can create an issue (either a bug report or a feature request based on the outcome of this step) and I can take a look at that accordingly!

**Reply by Firerouge** [reply]
This does work. To summarize what to do. 
Sign into the google account on a browser, I used firefox (non snap version)
Create a file `/etc/yt-dlp/config.txt` containing a single line `--cookies-from-browser firefox`

Then call ytmdl with the following `ytmdl --ytdl-config /etc/yt-dlp/config.txt https://music.youtube.com/playlist?list=LM`

**Comment by Firerouge** [comment]
Maybe it's a known issue, but errors from missing videos in playlists don't seem to be handled gracefully.

A playlist including, for example, this missing song https://music.youtube.com/watch?v=N_fGELxp10w fails and aborts with subsequent items in the playlist 
```
Traceback (most recent call last):
  File "/home/user/.local/bin/ytmdl", line 8, in <module>
    sys.exit(entry())
  File "/home/user/.local/lib/python3.10/site-packages/ytmdl/main.py", line 614, in entry
    extract_data()
  File "/home/user/.local/lib/python3.10/site-packages/ytmdl/main.py", line 607, in extract_data
    main(args)
  File "/home/user/.local/lib/python3.10/site-packages/ytmdl/main.py", line 238, in main
    song_name, verify_name = extract_song_name(args)
  File "/home/user/.local/lib/python3.10/site-packages/ytmdl/main.py", line 530, in extract_song_name
    song_name, verify_title = yt.get_title(args.url, args.ytdl_config)
  File "/home/user/.local/lib/python3.10/site-packages/ytmdl/yt.py", line 409, in get_title
    title = __get_title_from_yt(url, ytdl_config)
  File "/home/user/.local/lib/python3.10/site-packages/ytmdl/yt.py", line 368, in __get_title_from_yt
    return stringutils.remove_yt_words(data["title"])
TypeError: 'NoneType' object is not subscriptable
```

This is the case even with the following flags, `--download-archive dl.txt --quiet --on-meta-error youtube --ignore-errors --nolocal`
It's also the case with the following flags included in the yt-dlp config `--no-abort-on-error --ignore-errors`

**Reply by deepjyoti30** [reply]
@Firerouge Thanks for the issue, seems like this is a bug. I will open an issue for this.
